packages/compass-editor/package.json

```diff
@@ -77,6 +77,7 @@
     "@lezer/highlight": "^1.1.3",
     "@mongodb-js/compass-components": "^1.10.0",
     "@mongodb-js/mongodb-constants": "^0.6.0",
+    "mongodb-query-parser": "^2.5.0",
```

Comment:
Just a structural note here – I think, right now, the `...Editor` components exported by this package are content-agnostic, i.e. could be used for any kind of “code” text input, not necessarily JSON or JS input, and this package doesn’t really have dependencies on mongodb-specific logic.

The exception to the latter is `@mongodb-js/mongodb-constants`, which is used for autocompletion, but even there, it is (afaict) only being used in the `create...Autocompleter` exports, so the user of this package has to create an autocompleter using those functions, then pass them to the relevant Editor instance.

I think ideally it would be a bit cleaner to keep following this pattern and not make `BaseEditor` carry MongoDB-specific logic responsibility. What do you think? (Also maybe @gribnoysup has thoughts here as well…)

Reply:
Yeah, to be clear, I’m not saying not to use `mongodb-query-parser`; I’m saying, this validation should probably not be active unconditionally for anything that uses `BaseEditor` (this is where I think @gribnoysup is most likely to disagree with me, hence the ping :smile:), and it might be nice if the check functionality could be injected in a way that’s similar to how the autocomplete functionality is being injected by the user of the Editor components

Reply:
No, you're right @addaleax, this shouldn't be part of the base editor or editor package in general, this is too specific to query bar